include/rmm/cuda_device.hpp

```diff
@@ -38,6 +38,28 @@ struct cuda_device_id {
   /// @briefreturn{The wrapped integer value}
   [[nodiscard]] constexpr value_type value() const noexcept { return id_; }
 
+  /**
+   * @brief Equality comparison operator
+   *
+   * @param other The other `cuda_device_id` to compare to
+   * @return true if the two `cuda_device_id`s wrap the same integer value, false otherwise
+   */
+  [[nodiscard]] constexpr bool operator==(cuda_device_id const& other) const noexcept
+  {
+    return value() == other.value();
+  }
```

Comment:
https://jacquesheunis.com/post/hidden-friend-compilation/

```suggestion
private:
  [[nodiscard]] friend constexpr bool operator==(cuda_device_id const& lhs, cuda_device_id const& rhs) const noexcept
  {
    return lhs.value() == rhs.value();
  }
```

Reply:
None of the examples I can see on the web of hidden friends makes them private. Is that recommended?

Reply:
that was my bad, it actually doesn't matter if they are private or not, it doesn't change anything.